glotaran/project/project.py

```diff
@@ -0,0 +1,468 @@
+        else:
+            file = folder / PROJECT_FILE_NAME
+
+        if not file.exists():
+            if not create_if_not_exist:
+                raise FileNotFoundError(f"Project file {file} does not exists.")
+            Project.create(folder)
+
+        project_dict = load_dict(file, True)
+        project_dict["file"] = file
+        project_dict["folder"] = folder
+        return cls(**project_dict)
+
+    @property
+    def has_data(self) -> bool:
+        """Check if the project has datasets.
+
+        Returns
+        -------
+        bool
+            Whether the project has datasets.
+        """
+        return not self._data_registry.empty
+
+    @property
+    def data(self) -> dict[str, str]:
+        """Get all project datasets.
+
+        Returns
+        -------
+        dict[str, str]
+            The models of the datasets.
+        """
+        return self._data_registry.items
+
+    def load_data(self, name: str) -> xr.Dataset | xr.DataArray:
+        """Load a dataset.
+
+        Parameters
+        ----------
+        name : str
+            The name of the dataset.
+
+        Returns
+        -------
+        Result
+            The loaded dataset.
+
+        Raises
+        ------
+        ValueError
+            Raised if the dataset does not exist.
+        """
+        try:
+            return self._data_registry.load_item(name)
+        except ValueError:
+            raise ValueError(f"Dataset '{name}' does not exist.")
+
+    def import_data(self, path: str | Path, name: str | None = None):
+        """Import a dataset.
+
+        Parameters
+        ----------
+        path : str | Path
+            The path to the dataset.
+        name : str | None
+            The name of the dataset.
+        """
+        self._data_registry.import_data(path, name=name)
+
+    @property
+    def has_models(self) -> bool:
+        """Check if the project has models.
+
+        Returns
+        -------
+        bool
+            Whether the project has models.
+        """
+        return not self._model_registry.empty
+
+    @property
+    def models(self) -> dict[str, str]:
+        """Get all project models.
+
+        Returns
+        -------
+        dict[str, str]
+            The models of the project.
+        """
+        return self._model_registry.items
+
+    def load_model(self, name: str) -> Model:
+        """Load a model.
+
+        Parameters
+        ----------
+        name : str
+            The name of the model.
+
+        Returns
+        -------
+        Model
+            The loaded model.
+
+        Raises
+        ------
+        ValueError
+            Raised if the model does not exist.
+        """
+        try:
+            return self._model_registry.load_item(name)
+        except ValueError:
+            raise ValueError(f"Model '{name}' does not exist.")
+
+    def generate_model(
+        self,
+        name: str,
+        generator: str,
```

Comment:
Done.

About instances: This would be rather useless. If a user has his own generator which is not available via `generator_name`, then she should be able write mode files herself anyway. I added convenience function to get a project's model and data directory to aid with that. The other way would be to register her generator as plugin, but for this we need a plugin system for generators which is way out of this PR's scope.